src/microbe_stage/systems/MicrobeAISystem.cs

```diff
@@ -858,10 +896,21 @@ private void FleeFromPredators(ref WorldPosition position, ref MicrobeAI ai, ref
     }
 
     private void EngagePrey(ref MicrobeAI ai, ref MicrobeControl control, ref OrganelleContainer organelles,
-        ref WorldPosition position, CompoundBag ourCompounds, in Entity entity, Vector3 target, bool engulf,
-        float speciesAggression, float speciesFocus, float speciesActivity, float strain, Random random)
+        ref WorldPosition position, ref CompoundStorage compoundStorage, ref Health health, in Entity entity,
+        Vector3 target, bool engulf, float speciesAggression, float speciesFocus, float speciesActivity,
+        float strain, Random random, float atpLevel)
     {
-        control.SetStateColonyAware(entity, engulf ? MicrobeState.Engulf : MicrobeState.Normal);
+        var ourCompounds = compoundStorage.Compounds;
+
+        if (atpLevel <= Constants.ENGULF_NO_ATP_TRIGGER_THRESHOLD)
+        {
+            control.EnterEngulfModeForcedState(ref health, ref compoundStorage, entity, Compound.ATP);
```

Comment:
@hhyyrylainen Here's the line that calls `EnterEngulfModeForcedState` depending on ATP level.
I've also observed a cell enter engulf mode while being damaged by lack of ATP.

Reply:
I see there's now this call so the PR had been updated since I last looked.

BTW this method is not necessarily the best one call as there is a method that enters engulf mode normally or in a forced manner if required (and doesn't enter the mode if the cell would die from the damage).